apis/python/src/tiledb/vector_search/index.py

```diff
@@ -126,6 +126,13 @@ def __init__(
         self.thread_executor = futures.ThreadPoolExecutor()
 
     def query(self, queries: np.ndarray, k, **kwargs):
+        if queries.ndim != 1 and queries.ndim != 2:
```

Comment:
I think that we currently just support 2 dimensional query matrix. This is why one of your tests is failing.

Reply:
Hmm, we have some unit tests that query with a 1 dimensional query, i.e. here:
```
test_ingestion.py -> test_ivf_flat_ingestion_fvec()
    # Test single query vector handling
    _, result1 = index.query(query_vectors[10], k=k, nprobe=nprobe)
    assert accuracy(result1, np.array([gt_i[10]])) > MINIMUM_ACCURACY
```
Looking more into it, it seems that `IVF_FLAT` can support this, but not `FLAT`.

Reply:
Hmm, we have some unit tests that query with a 1 dimensional query, i.e. here:
```
test_ingestion.py -> test_ivf_flat_ingestion_fvec()
    # Test single query vector handling
    _, result1 = index.query(query_vectors[10], k=k, nprobe=nprobe)
    assert accuracy(result1, np.array([gt_i[10]])) > MINIMUM_ACCURACY
```
Looking more into it, it seems that `IVF_FLAT` can support this, but not `FLAT`.

Is this a scenario we want to support? If not then I can change this check and remove the test cases that use a 1 dimensional query. cc @lums658 / @ihnorton